src/cpp/include/nodegui/QtWidgets/QTreeWidget/ntreewidget.hpp

```diff
@@ -20,5 +21,35 @@ class NTreeWidget : public QTreeWidget, public NodeWidget {
       Napi::HandleScope scope(env);
       this->emitOnNode.Call({Napi::String::New(env, "itemSelectionChanged")});
     });
+
+    QObject::connect(
+        this, &QTreeWidget::itemClicked,
+        [=](QTreeWidgetItem* item, int column) {
+          Napi::Env env = this->emitOnNode.Env();
+          Napi::HandleScope scope(env);
+
+          // Disable deletion of the native instance for these by passing true
+          Napi::Object itemWrap = QTreeWidgetItemWrap::constructor.New(
+              {Napi::External<QTreeWidgetItem>::New(env, item),
+               Napi::Boolean::New(env, true)});
+
+          this->emitOnNode.Call({Napi::String::New(env, "itemClicked"),
+                                 itemWrap, Napi::Value::From(env, column)});
+        });
+
+    QObject::connect(
+        this, &QTreeWidget::itemChanged,
+        [=](QTreeWidgetItem* item, int column) {
+          Napi::Env env = this->emitOnNode.Env();
+          Napi::HandleScope scope(env);
+
+          // Disable deletion of the native instance for these by passing true
+          Napi::Object itemWrap = QTreeWidgetItemWrap::constructor.New(
```

Comment:
same here as above. We can allow deletion.

Reply:
Will change.